datamanager/src/main/java/eu/arrowhead/core/datamanager/DataManagerController.java

```
@@ -0,0 +1,587 @@
+	@ResponseBody public String historianSystemPut(
+			@PathVariable(value="systemName", required=true) String systemName,
+			@RequestBody String requestBody
+		) {
+		System.out.println("DataManager:PUT:Historian/"+systemName);
+
+		JsonParser parser= new JsonParser();
+		JsonObject obj = null;
+		try {
+			obj = parser.parse(requestBody).getAsJsonObject();
+		} catch(Exception je){
+			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, "malformed request");
+		}
+
+		String op = obj.get("op").getAsString();
+		if(op.equals("list")) {
+			//System.out.println("OP: list");
+			ArrayList<String> services = HistorianService.getServicesFromSystem(systemName);
+			//for (String srv: services) {
+			//  System.out.println(":" +srv);
+			//}
+			Gson gson = new Gson();
+			JsonObject answer = new JsonObject();
+			JsonElement servicelist = gson.toJsonTree(services);
+			answer.add("services", servicelist);
+			String jsonStr = gson.toJson(answer);
+			System.out.println(jsonStr);
+
+			return jsonStr;
+		} else if(op.equals("create")){
+			//System.out.println("OP: CREATE");
+			String srvName = obj.get("srvName").getAsString();
+			String srvType = obj.get("srvType").getAsString();
+			//System.out.println("Create SRV: "+srvName+" of type: "+srvType+" for: " + systemName);
+
+			/* check if service already exists */
+			ArrayList<String> services = HistorianService.getServicesFromSystem(systemName);
+			for (String srv: services) {
+				if(srv.equals(srvName)){
+					logger.info("  service:" +srv + " already exists");
+					Gson gson = new Gson();
+					JsonObject answer = new JsonObject();
+					answer.addProperty("createResult", "Already exists");
+					String jsonStr = gson.toJson(answer);
+					throw new ResponseStatusException(org.springframework.http.HttpStatus.CONFLICT, jsonStr);
+				}
+			}
+
+			/* create the service */
+			boolean ret = HistorianService.addServiceForSystem(systemName, srvName, srvType);
```

static problem
don't need a boolean return value, you should just throw an exception in the addServiceForSystem if an error occurs.